Compile error with g++-14, field.cpp.

I tried to compile feature/1.8/main with g++-14.

I got one compilation error:

compile: /export/home/guo/c_1.8/casual-1.8/middleware/buffer/source/field.cpp
error:
/export/home/guo/c_1.8/casual-1.8/middleware/buffer/source/field.cpp: In function ‘int casual::buffer::field::{anonymous}::explore::existence(const char*, casual::buffer::field::item_type, casual::buffer::field::size_type)’:
/export/home/guo/c_1.8/casual-1.8/middleware/buffer/source/field.cpp:574:42: error: ignoring return value of ‘constexpr std::vector<_Tp, _Alloc>::const_reference std::vector<_Tp, _Alloc>::at(size_type) const [with _Tp = long int; _Alloc = std::allocator<long int>; const_reference = const long int&; size_type = long unsigned int]’, declared with attribute ‘nodiscard’ [-Werror=unused-result]
  574 |                   buffer.index.at( id).at( occurrence);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
In file included from /usr/include/c++/14/vector:66,
                 from ../common/include/casual/platform.h:58,
                 from include/casual/buffer/internal/field.h:11,
                 from /export/home/guo/c_1.8/casual-1.8/middleware/buffer/source/field.cpp:9:
/usr/include/c++/14/bits/stl_vector.h:1199:7: note: declared here
 1199 |       at(size_type __n) const
      |       ^~
cc1plus: all warnings being treated as errors

aborting due to errors

The affected code is:

```
               // Just to force an exception if not present
               buffer.index.at( id).at( occurrence);

```

The value returned by at( occurence) is obviously discarded. I suspect that the [nodiscard] attribute on the method is new with g++-14.

The first change I thought of is to change to:
`               (void)buffer.index.at( id).at( occurrence);
`
but that is very "C-ish". Other possibilities are
`            std::ignore = buffer.index.at( id).at( occurrence);`
or
`            [[maybe_unused]] auto dummy = buffer.index.at( id).at( occurrence);`

(See notes in [cppreference](https://en.cppreference.com/w/cpp/utility/tuple/ignore))

With the std::ignore solution, it is possible that a "#include<tuple>" should be added as this is the header that defines std::ignore. Seems as if some other header already includes it but not depending on that may be nice.
 
I do not know which solution is the "Casual style" :-)  There might be better/nicer solutions :-)

--- a/middleware/buffer/source/field.cpp
+++ b/middleware/buffer/source/field.cpp
@@ -38,7 +38,7 @@
 #include <regex>
 #include <iostream>
 #include <sstream>
-
+#include <tuple>
 namespace casual
 {
    namespace buffer::field
@@ -571,7 +571,7 @@ namespace casual
                   const auto& buffer = pool_type::pool().get( common::buffer::handle::type{ handle});
 
                   // Just to force an exception if not present
-                  buffer.index.at( id).at( occurrence);
+                  std::ignore = buffer.index.at( id).at( occurrence);
 
                   return CASUAL_FIELD_SUCCESS;
                }

